dataflow/snippets/write_pubsub.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# [START dataflow_pubsub_write_with_attributes]
+import apache_beam as beam
+from apache_beam.io import PubsubMessage
+from apache_beam.io import WriteToPubSub
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+def item_to_message(item):
+    attributes = {}
+    attributes['buyer'] = item['name']
+    attributes['timestamp'] = str(item['ts'])
+    data = bytes(item['product'], 'utf-8')
+
+    return PubsubMessage(data=data, attributes=attributes)
+
+
+def write_to_pubsub(argv=None):
+
+    # Parse the pipeline options passed into the application.
+    class MyOptions(PipelineOptions):
+        @classmethod
+        # Define custom pipeline options that specify the project ID and Pub/Sub
+        # topic.
+        def _add_argparse_args(cls, parser):
+            parser.add_argument("--project", required=True)
+            parser.add_argument("--topic", required=True)
+
+    example_data = [
+        {'name': 'Robert', 'product': 'TV', 'ts': 1613141590000},
+        {'name': 'Maria', 'product': 'Phone', 'ts': 1612718280000},
+        {'name': 'Juan', 'product': 'Laptop', 'ts': 1611618000000},
+        {'name': 'Rebeca', 'product': 'Video game', 'ts': 1610000000000}
+    ]
+    options = MyOptions()
+
+    topic_path = f'projects/{options.project}/topics/{options.topic}'
```

Comment:
We could use [`pubsub.topic_path`](https://cloud.google.com/python/docs/reference/pubsub/latest/google.cloud.pubsub_v1.publisher.client.Client#google_cloud_pubsub_v1_publisher_client_Client_topic_path) here.

Reply:
This is using the Dataflow pipeline options pattern, and Dataflow is handling all the Pub/Sub interactions, so typically the pipeline code wouldn't take a Pub/Sub client dependency here

https://beam.apache.org/documentation/programming-guide/#creating-custom-options

Reply:
Sounds good, thanks for the clarification